plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/maven2/metadata/Maven2Metadata.java

```diff
@@ -0,0 +1,390 @@
+    checkNotNull(artifactId, "artifactId");
+    checkNotNull(prefix, "prefix");
+    return new Plugin(artifactId, prefix, Strings.isNullOrEmpty(name) ? artifactId : name);
+  }
+
+  public static Maven2Metadata newGroupLevel(final DateTime lastUpdated,
+                                             final String groupId,
+                                             final List<Plugin> plugins)
+  {
+    checkNotNull(lastUpdated);
+    checkNotNull(groupId);
+    return new Maven2Metadata(Level.GROUP, lastUpdated, groupId, null, null, plugins, null, null);
+  }
+
+  public static Maven2Metadata newArtifactLevel(final DateTime lastUpdated,
+                                                final String groupId,
+                                                final String artifactId,
+                                                final String latest,
+                                                final String release,
+                                                final Iterable<String> versions)
+  {
+    checkNotNull(lastUpdated);
+    checkNotNull(groupId);
+    checkNotNull(artifactId);
+    checkNotNull(latest);
+    checkNotNull(versions);
+    final BaseVersions bvs = new BaseVersions(latest, release, versions);
+    return new Maven2Metadata(Level.ARTIFACT, lastUpdated, groupId, artifactId, null, null, bvs, null);
+  }
+
+  public static Snapshot newSnapshot(final DateTime lastUpdated,
+                                     final String extension,
+                                     @Nullable final String classifier,
+                                     final String version)
+  {
+    checkNotNull(lastUpdated, "lastUpdated");
+    checkNotNull(extension, "extension");
+    checkNotNull(version, "version");
+    return new Snapshot(lastUpdated, extension, classifier, version);
+  }
+
+  public static Maven2Metadata newVersionLevel(final DateTime lastUpdated,
+                                               final String groupId,
+                                               final String artifactId,
+                                               final String version,
+                                               final long snapshotTimestamp,
+                                               final int snapshotBuildNumber,
+                                               final List<Snapshot> snapshots)
+  {
+    checkNotNull(lastUpdated);
+    checkNotNull(groupId);
+    checkNotNull(artifactId);
+    checkNotNull(version);
+    checkArgument(snapshotTimestamp > 0, "snapshotTimestamp: %s", snapshotTimestamp);
+    checkArgument(snapshotBuildNumber > 0, "snapshotBuildNumber: %s", snapshotBuildNumber);
+    List<Snapshot> ss = Lists.newArrayList();
+    if (snapshots != null) {
+      ss.addAll(snapshots);
+    }
+    final Snapshots snaps = new Snapshots(snapshotTimestamp, snapshotBuildNumber, ss);
+    return new Maven2Metadata(Level.BASEVERSION, lastUpdated, groupId, artifactId, version, null, null, snaps);
+  }
+}
```

Comment:
general comment: well-defined classes in their own source file can be easier to read/grok compared to a long list of somewhat related classes collected in a single file.